### PR TITLE
Lc/deserialize

### PR DESCRIPTION
**Pull request by loucaponi**
Adds ability to deserialize transaction types of -4 (sidechain)

### CODE REVIEW

File: src/transaction.js [hunk 1]
```diff
@@ -316,6 +307,12 @@ function deserializeTx (hexStr: string, withPrevScriptPubKey: boolean = false):
     offset = newOffset;
   }
 
+  if (txObj.version === -4) {
```
Comment (yurikoinaba) [inline src/transaction.js:310]: Can we add `-4` as a constant in the constants file?

File: test/helper/formatSidechain.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  var txObj = {};
+  txObj.version = txJson.version;
+  txObj.locktime = txJson.locktime;
+
+  txObj.vsc_ccout = txJson.vsc_ccout.map(i => {
+    delete i.scid; // RPC call would be needed for scid
+    return { 
+      ...i, 
+      value: i.value * 1e8, 
+      ftScFee: i.ftScFee * 1e8 ,
```
Comment (yurikoinaba) [inline test/helper/formatSidechain.js:11]: Nice, I like the `1e8` better than `100000000` I used in the `formatCertificate` function. Do you mind changing it there too for consistency?

Comment (yurikoinaba) [reply test/helper/formatSidechain.js:11]: Also nitpick - there is a space before the `,`. Would love to get a linter or prettier in this repo at some point :)

File: test/helper/formatSidechain.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+  const formattedOutput = [];
+  txJson.vout.forEach(output => {
+      formattedOutput.push({
+          script: output.scriptPubKey.hex,
+          satoshis: output.valueZat,
+          ...(output['backward transfer'] && {
+              isFromBackwardTransfer: output['backward transfer'], 
+              pubKeyHash: output.pubkeyhash
+          })
```
Comment (yurikoinaba) [inline test/helper/formatSidechain.js:59]: Backward transfers are only returned in certificates, so you should be ok to delete these lines.

**Review by yurikoinaba — COMMENTED** [review]
Nice job ✨

File: test/helper/formatSidechain.js (hunk 2 above)
Comment (loucaponi) [reply test/helper/formatSidechain.js:11]: yes, agree, linter is a good idea

**Review by paolocappelletti — APPROVED** [review]

**Review by yurikoinaba — APPROVED** [review]

File: src/types.js [hunk 4]
```diff
@@ -11,44 +11,47 @@ export type TXOBJ = {
...
-      vBitVectorCertificateFieldConfig: number[][],
-      forwardTransferScFee: number,
-      mainchainBackwardTransferScFee: number,
-      mbtrRequestDataLength: number
-    }[],
-    vft_ccout: {
-      scid: string,
-      satoshis: number,
-      address: string,
-      mcReturnAddress: string
```
Comment (otoumas) [inline src/types.js:33]: @loucaponi @MarcoOl94 This field has been removed intentionally? It is part of the vft_ccout right? At least in bitcore-lib-zen we are adding it 🤔

Comment (MarcoOl94) [reply src/types.js:33]: Yes this fhield should be kept.

File: src/sidechain.js [hunk 5]
```diff
@@ -0,0 +1,256 @@
...
+        const address = buf.slice(offset, offset + 32).reverse().toString('hex');
+        offset += 32;
+
+        const scid = buf.slice(offset, offset + 32).reverse().toString('hex');
+        offset += 32;
+
+        outputs.push({
+           value,
+           address,
+           scid
```
Comment (otoumas) [inline src/sidechain.js:204]: @loucaponi I think that `mcReturnAddress` is missing here as well 🤔

File: lib/sidechain.js [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+    var value = readUInt64LE(buf, offset);
+    offset += 8;
+    var address = buf.slice(offset, offset + 32).reverse().toString('hex');
+    offset += 32;
+    var scid = buf.slice(offset, offset + 32).reverse().toString('hex');
+    offset += 32;
+    outputs.push({
+      value: value,
+      address: address,
+      scid: scid
```
Comment (otoumas) [inline lib/sidechain.js:207]: @loucaponi I think `mcReturnAddress` is missing here?

**Pull request merged**